.github/workflows/lint.yml

```diff
@@ -0,0 +1,68 @@
+name: Lint
+#
+# This workflow lints the UI library and reports back any suggested fixes
+#
+
+on:
+    workflow_dispatch:
+    workflow_call:
+
+permissions:
+    contents: read
+    pull-requests: write
+
+defaults:
+    run:
+        shell: bash
+
+jobs:
+    # --- Lint pre-compiled assets for consistency --- #
+    lint:
+        name: Lint
+        runs-on: ubuntu-latest
+        timeout-minutes: 5
+        outputs:
+            eslint_diff_files: ${{ steps.changed-files.outputs.eslint_diff_files }}
+            stylelint_diff_files: ${{ steps.changed-files.outputs.stylelint_diff_files }}
+        steps:
+            # install but don't build - we're linting pre-compiled assets
+            - name: Check out code
+              uses: actions/checkout@v4
+              with:
+                  fetch-depth: 2
+
+            - name: Setup Job and Install Dependencies
+              uses: ./.github/actions/setup-job
+
+            - name: Check for changed files
+              id: changed-files
+              run: |
+                  echo "eslint=$(git diff --name-only --diff-filter=d -- "projects/documentation/**/*.ts" "packages/**/*.ts" "tools/**/*.ts" | xargs)" >> $GITHUB_OUTPUT
+                  echo "stylelint=$(git diff --name-only --diff-filter=d -- "packages/**/*.css" "tools/**/*.css" | xargs)" >> $GITHUB_OUTPUT
+
+            - name: echo eslint changed files
+              run: echo ${{ steps.changed-files.outputs.eslint }}
+
+            - name: echo stylelint changed files
+              run: echo ${{ steps.changed-files.outputs.stylelint }}
+
+            - name: Lint component styles
+              uses: reviewdog/action-stylelint@v1.30.0
```

Comment:
We use reviewdog in CSS to comment on pull requests if linting violations are found. It's awesome!

Reply:
I can comment this out if we prefer since it's new functionality. We use in CSS but it would be new to SWC.

Reply:
Let’s leave it in! It’s a two-way door decision - we can easily remove it in a blip if we change our minds. I’m in favor of keeping it! 🙌